crux_time/src/command.rs

```diff
@@ -0,0 +1,206 @@
+use std::{future::Future, marker::PhantomData};
+
+use crux_core::{command::RequestBuilder, Command, Request};
+use futures::{
+    channel::oneshot::{self, Sender},
+    select, FutureExt,
+};
+
+use crate::{get_timer_id, Duration, Instant, TimeRequest, TimeResponse, TimerId};
+
+pub struct Time<Effect, Event> {
+    // Allow impl level trait bounds to avoid repetition
+    effect: PhantomData<Effect>,
+    event: PhantomData<Event>,
+}
+
+impl<Effect, Event> Time<Effect, Event>
+where
+    Effect: Send + From<Request<TimeRequest>> + 'static,
+    Event: Send + 'static,
+{
+    pub fn now() -> RequestBuilder<Effect, Event, impl Future<Output = TimeResponse>> {
+        Command::request_from_shell(TimeRequest::Now)
+    }
+
+    pub fn notify_at(
+        instant: Instant,
+    ) -> (
+        RequestBuilder<Effect, Event, impl Future<Output = TimeResponse>>,
+        TimerHandle,
+    ) {
+        let id = get_timer_id();
+        let (sender, mut receiver) = oneshot::channel();
+
+        let builder = RequestBuilder::new(move |ctx| async move {
+            // We loop so that we can ignore the Err case of the oneshot receiver and still wait for the original request
```

Comment:
What are the semantics here? If the sender is dropped, will the `select!` still block on the receiver (never resolving) or will the receiver immediately resolve to `Err(_)`, causing a busy loop?

Reply:
From some testing I've done, dropping the `Sender` does not wake the future, so I think the `Err` branch of the cancel channel select is actually unreachable 🤔 I think it's because the `Receiver` is a `FusedFuture` and slect checks it's termination status.

Might change the code to remove the loop and put `unreachable` in there